arrow-core-data/src/main/kotlin/arrow/core/NonEmptyList.kt

```diff
@@ -703,3 +697,9 @@ private class NonEmptyListShow<A>(
 ) : Show<NonEmptyList<A>> {
   override fun NonEmptyList<A>.show(): String = show(SA)
 }
+
+fun <A, B, Z> NonEmptyList<A>.zip(fb: NonEmptyList<B>, f: (A, B) -> Z): NonEmptyList<Z> =
+  zip(fb).map { ab: Pair<A, B> -> f(ab.first, ab.second) }
+
+fun <A, B> NonEmptyList<A>.zip(fb: NonEmptyList<B>): NonEmptyList<Pair<A, B>> =
+  NonEmptyList(Pair(head, fb.head), tail.zip(fb.tail).map { Pair(it.first, it.second) })
```

Comment:
```suggestion
fun <A, B, Z> NonEmptyList<A>.zip(fb: NonEmptyList<B>, f: (A, B) -> Z): NonEmptyList<Z> =
  NonEmptyList(f(head, fb.head), tail.zip(fb.tail, f))

fun <A, B> NonEmptyList<A>.zip(fb: NonEmptyList<B>): NonEmptyList<Pair<A, B>> =
  zip(fb, ::Pair)
```

Idem `Ior`, here it's worse since you'd introduce `Pair` for every element.